apps/OpenSpace/ext/launcher/src/sgctedit/windowcontrol.cpp

```diff
@@ -720,17 +749,55 @@ sgct::config::Window WindowControl::generateWindowInformation() const {
     vp.position = sgct::vec2(0.f, 0.f);
     vp.size = sgct::vec2(1.f, 1.f);
     vp.projection = generateProjectionInformation();
+    window.viewports.clear();
     window.viewports.push_back(vp);
     
     window.isDecorated = _windowDecoration->isChecked();
-    if (window.isFullScreen) {
-        window.monitor = _monitor->currentIndex();
-    }
-
     if (!_windowName->text().isEmpty()) {
         window.name = _windowName->text().toStdString();
     }
-    return window;
+}
+
+void WindowControl::setProjectionPlanar(float hfov, float vfov) {
+    _planar.fovH->setValue(hfov);
+    _planar.fovV->setValue(vfov);
+    _projectionType->setCurrentIndex(static_cast<int>(ProjectionIndices::Planar));
+}
+
+void WindowControl::setProjectionFisheye(int quality, bool spoutOutput) {
+    setQualityComboBoxFromLinesResolution(quality, _fisheye.quality);
+    _fisheye.spoutOutput->setChecked(spoutOutput);
+    _projectionType->setCurrentIndex(static_cast<int>(ProjectionIndices::Fisheye));
+}
+
+void WindowControl::setProjectionSphericalMirror(int quality) {
+    setQualityComboBoxFromLinesResolution(quality, _sphericalMirror.quality);
+    _projectionType->setCurrentIndex(
+        static_cast<int>(ProjectionIndices::SphericalMirror)
+    );
+}
+
+void WindowControl::setProjectionCylindrical(int quality, float heightOffset) {
+    setQualityComboBoxFromLinesResolution(quality, _cylindrical.quality);
+    _cylindrical.heightOffset->setValue(heightOffset);
+    _projectionType->setCurrentIndex(static_cast<int>(ProjectionIndices::Cylindrical));
+}
+
+void WindowControl::setProjectionEquirectangular(int quality, bool spoutOutput) {
+    setQualityComboBoxFromLinesResolution(quality, _equirectangular.quality);
+    _equirectangular.spoutOutput->setChecked(spoutOutput);
+    _projectionType->setCurrentIndex(
+        static_cast<int>(ProjectionIndices::Equirectangular)
+    );
+}
+
+void WindowControl::setQualityComboBoxFromLinesResolution(int lines, QComboBox* combo) {
+    for (unsigned int v = 0; v < nQualityTypes; ++v) {
```

Comment:
Would put in a:
`ghoul_assert(combo, "Invalid pointer")`
in the beginning to be sure